middleware/proxy/lookup.go

```diff
@@ -108,11 +108,11 @@ func (p Proxy) lookup(state request.Request) (*dns.Msg, error) {
 
 			atomic.AddInt64(&host.Conns, 1)
 
-			reply, backendErr := upstream.Exchanger().Exchange(context.TODO(), host.Name, state)
+			reply, _ := upstream.Exchanger().Exchange(context.TODO(), host.Name, state)
```

Comment:
`backendErr` was only used to determine if the reply is valid
~~~ Go
if backendErr == nil {
    return reply, nil
}
~~~

I changed the `Exchange` behavior to use reply instead, so the (dnstap) error isn't fatal:

~~~ Go
// When *dns.Msg is not nil it is valid.
// When both *dns.Msg and error are not nil, the error should be reported.
func (d *dnsEx) Exchange(ctx context.Context, addr string, state request.Request) (*dns.Msg, error) {
~~~